packages/big-design/src/components/Worksheet/spec.tsx

```diff
@@ -85,19 +95,23 @@ let handleErrors = jest.fn();
 const initialStoreState = useStore.getState();
 
 beforeEach(() => {
-  handleChange = jest.fn();
-  handleErrors = jest.fn();
-  act(() => useStore.setState(initialStoreState, true));
+  act(() => {
+    handleChange = jest.fn();
+    handleErrors = jest.fn();
+    useStore.setState(initialStoreState, true);
+  });
 });
 
-test('renders worksheet', () => {
+test('renders worksheet', async () => {
   const { container } = render(<Worksheet columns={columns} items={items} onChange={handleChange} />);
 
   expect(container.firstChild).toMatchSnapshot();
+
+  await waitForElement(() => screen.getAllByRole('combobox'));
```

Comment:
🤔 What is this line for?

Reply:
From what I've gathered, the Select component performs some state changes when rendering so we actually need to wait for it to be done with the updates else we get an error due to state changes not wrapped in an act.

Reply:
🧠 TIL - thanks 😃